src/hospital.c

```diff
@@ -0,0 +1,568 @@
+        hospital->waiting_list[ward_type] = malloc( sizeof (list) );
+        initialise_list( hospital->waiting_list[ward_type] );
+
+        for( hcw_type = 0; hcw_type < N_WORKER_TYPES; hcw_type++ )
+            hospital->n_workers[hcw_type] += params->n_wards[ward_type] * params->n_hcw_per_ward[ward_type][hcw_type];
+        
+        for( ward_idx = 0; ward_idx < params->n_wards[ward_type]; ward_idx++ )
+            initialise_ward( &(hospital->wards[ward_type][ward_idx]),
+                             ward_idx, ward_type,
+                             params->n_ward_beds[ward_type],
+                             params->n_hcw_per_ward[ward_type][DOCTOR],
+                             params->n_hcw_per_ward[ward_type][NURSE] );
+    }
+}
+
+/*****************************************************************************************
+*  Name:		set_up_hospital_networks
+*  Description: for all hospitals in simulation calls setup functions for related networks
+*  Returns:		void
+******************************************************************************************/
+void set_up_hospital_networks( model *model )
+{
+    int hospital_idx, hcw_idx, n_healthcare_workers, ward_idx, ward_type;
+    long *healthcare_workers;
+    hospital *hospital;
+
+    for( hospital_idx = 0; hospital_idx < model->params->n_hospitals; hospital_idx++ )
+    {
+        hospital = &model->hospitals[hospital_idx];
+        //Setup hospital workplace network.
+        n_healthcare_workers = 0;
+        healthcare_workers = calloc( hospital->n_workers[DOCTOR] + hospital->n_workers[NURSE], sizeof(long) );
+
+        //Setup HCW-patient networks for all wards.
+        for ( ward_type = 0; ward_type < N_HOSPITAL_WARD_TYPES; ward_type++ )
+        {
+            for( ward_idx = 0; ward_idx < hospital->n_wards[ward_type]; ward_idx++ )
+            {
+                for( hcw_idx = 0; hcw_idx < hospital->wards[ward_type][ward_idx].n_worker[DOCTOR]; hcw_idx++ )
+                    healthcare_workers[n_healthcare_workers++] = hospital->wards[ward_type][ward_idx].doctors[hcw_idx].pdx;
+
+                for( hcw_idx = 0; hcw_idx < hospital->wards[ward_type][ward_idx].n_worker[NURSE]; hcw_idx++ )
+                    healthcare_workers[n_healthcare_workers++] = hospital->wards[ward_type][ward_idx].nurses[hcw_idx].pdx;
+
+                set_up_ward_networks( &(hospital->wards[ward_type][ward_idx]), model->params->max_hcw_daily_interactions );
+            }
+        }
+
+        //Setup HCW workplace network.
+        hospital->hospital_workplace_network = calloc( 1, sizeof( network ));
+        hospital->hospital_workplace_network = new_network( n_healthcare_workers, HOSPITAL_WORK );
+
+        build_watts_strogatz_network( hospital->hospital_workplace_network, n_healthcare_workers, model->params->hcw_mean_work_interactions, model->params->work_network_rewire, TRUE );
+        relabel_network( hospital->hospital_workplace_network, healthcare_workers );
```

Comment:
List of healthcare workers, ordered by ward number, set true to false, when the network is set up, it will keep the structure

Reply:
Maybe look into a separate rewire probability for these networks